tiledb/sm/serialization/fragment_metadata.cc

```diff
@@ -165,70 +165,110 @@ Status fragment_metadata_from_capnp(
   frag_meta->tile_index_base() = frag_meta_reader.getTileIndexBase();
   frag_meta->version() = frag_meta_reader.getVersion();
 
+  // Set the array schema and most importantly retrigger the build
+  // of the internal idx_map. Also set array_schema_name which is used
+  // in some places in the global writer
+  frag_meta->set_array_schema(array_schema);
+  frag_meta->set_schema_name(array_schema->name());
+  frag_meta->set_dense(array_schema->dense());
+
   FragmentMetadata::LoadedMetadata loaded_metadata;
+
+  // num_dims_and_attrs() requires a set array schema, so it's important
+  // schema is set above on the fragment metadata object.
+  uint64_t num_dims_and_attrs = frag_meta->num_dims_and_attrs();
+
+  // This field may not be present here in some usecases such as refactored
+  // query, but readers on the server side require this vector to have the first
+  // dimension properly allocated when loading its data on demand.
```

Comment:
Small change... I'd like to create a function that does the following three lines in the fragment metadata... Maybe call it resize_tile_offsets_vectors. We could also get rid of the tile_offsets_mtx() accessor, which is really weird to give access to the outside of the class. This could be done for the tile_var_offsets_ and tile_var_sizes below as well.